tests-js/loan-token/TradingTestToken.test.js

```diff
@@ -0,0 +1,306 @@
+				sovryn,
+				loanToken,
+				set_demand_curve,
+				lend_to_pool,
+				open_margin_trade_position,
+				priceFeeds,
+				true,
+				RBTC,
+				WRBTC,
+				SUSD,
+				accounts
+			);
+			await close_partial_margin_trade(
+				sovryn,
+				loanToken,
+				set_demand_curve,
+				lend_to_pool,
+				open_margin_trade_position,
+				priceFeeds,
+				false,
+				RBTC,
+				WRBTC,
+				SUSD,
+				accounts
+			);
+		});
+
+		it("Test close partial margin trade sov reward payment", async () => {
+			await close_partial_margin_trade_sov_reward_payment(
+				sovryn,
+				set_demand_curve,
+				lend_to_pool,
+				open_margin_trade_position,
+				true,
+				FeesEvents,
+				loanToken,
+				RBTC,
+				WRBTC,
+				SUSD,
+				SOV,
+				accounts
+			);
+		});
+
+		it("Test close partial margin trade sov reward payment false", async () => {
+			await close_partial_margin_trade_sov_reward_payment(
+				sovryn,
+				set_demand_curve,
+				lend_to_pool,
+				open_margin_trade_position,
+				false,
+				FeesEvents,
+				loanToken,
+				RBTC,
+				WRBTC,
+				SUSD,
+				SOV,
+				accounts
+			);
+		});
+
+		// verifies that the loan size is computed correctly
+		it("Test getMarginBorrowAmountAndRate", async () => {
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, accounts[0]);
+			const deposit = hunEth;
+			const borrowAmount = await loanToken.getMarginBorrowAmountAndRate(oneEth.mul(new BN(4)), deposit);
+			const monthly_interest = borrowAmount[1].mul(new BN(28)).div(new BN(365));
+			// divide by 1000 because of rounding
+			const actualAmount = borrowAmount[0].div(new BN(1000));
+			const expectedAmount = deposit.mul(new BN(4)).mul(hunEth).div(hunEth.sub(monthly_interest)).div(new BN(1000));
+			expect(actualAmount.eq(expectedAmount)).to.be.true;
+		});
+
+		// test the correct max escrow amount is returned (considering that the function is actually returning a bit less than the max)
+		it("Test getMaxEscrowAmount", async () => {
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, accounts[0]);
+
+			const maxEscrowAmount1x = await loanToken.getMaxEscrowAmount(oneEth);
+			const maxEscrowAmount4x = await loanToken.getMaxEscrowAmount(oneEth.mul(new BN(4)));
+			expect(maxEscrowAmount1x.eq(maxEscrowAmount4x.mul(new BN(4)))).to.be.true;
+		});
+
+		// test the correct max escrow amount is returned (considering that the function is actually returning a bit less than the max)
+		it("Test margin trading without early access token should fail", async () => {
+			const early_access_token = await TestToken.new("Sovryn Early Access Token", "SEAT", 1, 10);
+			await early_access_token.transfer(accounts[1], await early_access_token.balanceOf(accounts[0]));
+			await loanToken.setEarlyAccessToken(early_access_token.address);
+
+			expectRevert(
+				margin_trading_sending_loan_tokens(accounts, sovryn, loanToken, SUSD, RBTC, priceFeeds, false),
+				"No early access tokens"
+			);
+		});
+	});
+});
```

Comment:
test_increasing_position_of_other_trader_should_fail is missing

Reply:
added